cfgrammar/src/lib/yacc/mod.rs

```diff
@@ -14,6 +14,27 @@ pub use self::{
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum YaccKindResolver {
+    /// The user can specify `%grmtools` in their grammar but unless it is compatible with this `YaccKind`, it's an error
+    Force(YaccKind),
+    /// Use `YaccKind` if the user doesn't specify `%grmtools` in their grammar
+    Default(YaccKind),
+    /// The user must specify `%grmtools` in their grammars or we throw an error
+    NoDefault
+}
+
+impl YaccKindResolver {
+    pub(crate) fn self_describing(self) -> bool {
```

Comment:
Hmm, there is probably a better name for the like: `uses_grmtools_section`

Reply:
Nevermind, once clippy was done with these, they became one liners, and since they were only used in one place.
It is much easier to just inline them and remove the methods entirely.